lib/request.coffee

```diff
@@ -322,4 +306,53 @@ module.exports = getRequest = ({
 	# (or resolve to) a new response, or throw/reject.
 	###
 
+	###*
+	# @summary Refresh token on user request
+	# @function
+	# @public
+	#
+	# @description
+	# This function automatically refresh token on user request.
+	# The token is refreshed even if it is not expired yet.
+	# Error is thrown if user is not authorized.
+	#
+	# @param {String} options.url - relative url
+	#
+	# @returns {String} token - new token
+	#
+	# @example
+	# request.refreshToken
+	# 	baseUrl: 'https://api.resin.io'
+	###
+
+	exports.refreshToken = (options = {}) ->
+
+		{baseUrl} = options
```

Comment:
Nit: we'd normally have spaces around `baseUrl` here

Reply:
Ah, in fact: you don't need to take options at all.

If you squint a bit, this logic is inside the initial `getRequest` call, so the `options` wayyyyy above are in scope here too.

(Yes, defining structure with indenting makes this confusing, I agree, coffeescript is a mess)

Reply:
Without the options I've got `BaseUrl is not defined` error. It is defined only in `prepareOptions` method not `getRequest` scope.

Reply:
You're totally right - I'd assumed that this was a factory option, but as in our other chat, it's an option to the `send` call itself. Nevermind!